src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2297,7 +2297,7 @@ private function registerSchedulerConfiguration(ContainerBuilder $container, Php
         }
 
         // BC layer Scheduler < 7.3
-        if (!class_exists(SchedulerTriggerNormalizer::class)) {
+        if (!class_exists(SchedulerTriggerNormalizer::class) || !ContainerBuilder::willBeAvailable('symfony/serializer', DenormalizerInterface::class, ['symfony/framework-bundle', 'symfony/scheduler'])) {
```

Comment:
this needs to be done *before* the `class_exists(SchedulerTriggerNormalizer::class)` check, as this is what triggers the loading of the SchedulerTriggerNormalizer class with missing dependency.

Alternatively, use `$container->classExists()` which deals with such error (considering that the class does not exist if it misses its dependencies)